src/main/java/com/talentpath/shamazin/showItemPage/services/ItemFamilyService.java

```diff
@@ -35,5 +35,9 @@ public ItemFamily getItemFamilyById(Integer id) throws NoSuchItemFamilyException
 
     public ItemFamily addItemFamily(ItemFamily itemFamily){
         return dao.saveAndFlush(itemFamily);
-    };
+    }
+
+    public ItemFamily getItemFamilyByItemId(Integer itemId) {
+        return dao.findByItemId(itemId);
```

Comment:
I think you could add exception handling right here incase nothing is returned but other than that looks good